lib/fetch/request.js

```diff
@@ -415,25 +417,31 @@ class Request {
     }
 
     // 32. If init is not empty, then:
-    if (Object.keys(init).length !== 0) {
+    if (initHasKey) {
+      /** @type {HeadersList} */
+      const headersList = this[kHeaders][kHeadersList]
       // 1. Let headers be a copy of this’s headers and its associated header
       // list.
-      let headers = new Headers(this[kHeaders])
+      let headers
```

Comment:
this comment no longer matches what is happening. Instead, you can combine steps 1 & 2 into a single step and have both comments come before this:

```js
const headers = init.headers !== undefined ? init.headers : new HeadersList(headersList)
```

Reply:
Change with 47d95b793e479e3b1a8b3115aecef918716a060e